### DISCUSSION TITLE
Essential metabolic tasks verification

### DISCUSSION THREAD

**Discussion by Flash1593** [SysBioChalmers/Human-GEM#394] [Q&A] [answered]
For verification of the reference model (Human-GEM) for performing all the essential tasks after performing addBoundaryMets step, it is giving the following error -

>> checkTasks(ihuman,[],true,false,false,essentialTasks);
PASS: [ER] Aerobic rephosphorylation of ATP from glucose
Index exceeds the number of array elements. Index must not exceed 0.

Error in checkTasks (line 217)
    essentialFluxes(:,i) = sol.x(1:numel(model.rxns));
 
What should I do to troubleshoot the problem? Any help would be really appreciated.

**Comment by JonathanRob** [comment]
Hi @Flash1593, what version/branch of Human-GEM and RAVEN are you using? And if you could provide the command(s) run prior to this one, as well as their output, that would be helpful.

**Reply by Flash1593** [reply]
I had cloned both Human-GEM & RAVEN using github
git clone https://github.com/SysBioChalmers/Human-GEM.git
git clone --depth=1 https://github.com/SysBioChalmers/RAVEN.git

These are the commands and their output prior to essential task command

>> load('Human-GEM.mat')
>> ihuman=addBoundaryMets(ihuman);

Boundary metabolites were added to 1665 reactions.
New (boundary) versions of 1665 metabolites were added to the model.
>> essentialTasks=parseTaskList('metabolicTasks_Essential.xlsx')

essentialTasks = 

  57×1 struct array with fields:

    id
    description
    shouldFail
    printFluxes
    comments
    inputs
    LBin
    UBin
    outputs
    LBout
    UBout
    equations
    LBequ
    UBequ
    changed
    LBrxn
    UBrxn
>> checkTasks(ihuman,[],true,false,false,essentialTasks);
PASS: [ER] Aerobic rephosphorylation of ATP from glucose
Index exceeds the number of array elements. Index must
not exceed 0.

Error in checkTasks (line 217)
    essentialFluxes(:,i) = sol.x(1:numel(model.rxns));

I tried repeating the last command and it gave this

>> checkTasks(ihuman,[],true,false,false,essentialTasks);
PASS: [ER] Aerobic rephosphorylation of ATP from glucose
PASS: [ER] Aerobic rephosphorylation of GTP
PASS: [ER] Aerobic rephosphorylation of CTP
Index exceeds the number of array elements. Index must not exceed 0.

Error in checkTasks (line 217)
    essentialFluxes(:,i) = sol.x(1:numel(model.rxns));
 
>> checkTasks(ihuman,[],true,false,false,essentialTasks);
PASS: [ER] Aerobic rephosphorylation of ATP from glucose
Index exceeds the number of array elements. Index must not exceed 0.

Error in checkTasks (line 217)
    essentialFluxes(:,i) = sol.x(1:numel(model.rxns));


Is it a solver problem?

**Comment by JonathanRob** [comment, marked as answer]
You're correct, it looks like when using **glpk**, it results in an "Infeasible" solution and failure for many of the tasks. I'm not sure why this is the case, but I think glpk tends to have problems with larger problems/models. Could you try using another solver, such as **gurobi**?

**Reply by Flash1593** [reply]
Hi, I had changed the solver to gurobi for RAVEN toolbox but still I'm getting the same error.

**Reply by JonathanRob** [reply]
We fixed a [bug](https://github.com/SysBioChalmers/RAVEN/issues/425) in RAVEN yesterday that prevented users from changing the solver with `setRavenSolver` unless they restarted Matlab. Can you update to the latest version of RAVEN (released yesterday), restart your Matlab session, and try again with Gurobi?

If you're still getting the same problem after that, then let me know and we can work on a fix.

**Reply by Flash1593** [reply]
Yeah, I restarted Matlab & changed the solver to gurobi, now the checktask algorithm is working fine. Thanks for the help.

**Reply by JonathanRob** [reply]
Ok great, glad it's working now.
